src/routers/api.ts

```diff
@@ -231,40 +256,45 @@ router.post('/transfers', async (req, res, next) => {
     await blobsHandler.retrieveMetadata(req.body.path);
     let senderDestination: string | undefined = undefined;
     if (typeof req.body.sender === 'string') {
-      if (!req.body.sender.startsWith(peerID)) {
-        throw new RequestError('Invalid sender');
-      } else {
-        const destination = req.body.sender.substring(peerID.length + 1);
-        if(destination.length > 0) {
-          senderDestination = destination;
+      const segments = req.body.sender.split('/');
+      if (segments[0] !== peerID) {
+        throw new RequestError(`Sender ID mismatch expected=${peerID} recieved=${segments[0]}`, 400);
+      }
+      if (segments.length > 1) {
+        if (!config.destinations?.includes(segments[1])) {
+          throw new RequestError(`Unknown sender destination expected=${config.destinations?.join('|')} recieved=${segments[1]}`, 400);
         }
+        senderDestination = segments[1];
       }
     }
     let recipientID: string;
     let recipientDestination: string | undefined = undefined;
     if (typeof req.body.recipient === 'string') {
-      const index = req.body.recipient.indexOf(utils.constants.ID_SEGMENT_SEPARATOR);
-      if (index !== -1) {
-        recipientID = req.body.recipient.substring(0, index);
-        const destination = req.body.recipient.substring(index + 1);
-        if(destination.length > 0) {
-          recipientDestination = destination;
-        }
-      } else {
-        recipientID = req.body.recipient;
+      const segments = req.body.recipient.split('/');
+      recipientID = segments[0];
+      if (segments.length > 1) {
+        recipientDestination = segments[1];
       }
     } else {
       throw new RequestError('Missing recipient', 400);
     }
-    let recipientURL = config.peers.find(peer => peer.id === recipientID)?.endpoint;
-    if (recipientURL === undefined) {
-      throw new RequestError(`Unknown recipient`, 400);
+    let recipientEndpoint: string;
+    recipientEndpoint = config.p2p.endpoint ?? `https://${config.p2p.hostname}:${config.p2p.port}`;
```

Comment:
This looks wrong... maybe this was supposed to be inside the `if` below?

Reply:
Precisely